src/components/Comments.jsx

```diff
@@ -1,37 +1,93 @@
-      const [isLoading, setIsLoading] = useState(true)
+const Comments = ({ article_id, userId }) => {
+  const [comments, setComments] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [newComment, setNewComment] = useState("");
+  const [inputError, setInputError] = useState("");
+  const [submitStatus, setSubmitStatus] = useState("");
 
-    useEffect(() => {
-        const fetchComments = async () => {
-            const response = await getComments(article_id)
-            setComments(response.comments)
-            setIsLoading(false)
-        }
-        fetchComments()
-    }, [])
+  useEffect(() => {
+    const fetchComments = async () => {
+      const response = await getComments(article_id);
+      setComments(response.comments);
+      setIsLoading(false);
+    };
+    fetchComments();
+  }, []);
 
-    
-    if (isLoading === true){
-        return <p>Loading...</p>
-    }
-    else if (isLoading === true && comments.length === 0){
-      return(
-        <section className="comments-section">
-          <p>No comments yet...</p>
-        </section>
-      )
+  const handleChange = (e) => {
+    setNewComment(e.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    setInputError("");
+    setSubmitStatus("");
+    if (userId === "") {
+      setInputError("Please sign in to make a comment");
+    } else if (newComment === "") {
+      setInputError("Please input some text to make a comment");
+    } else {
+      const now = new Date().toISOString();
+      setComments((currentComments) => {
+        return [
+          {
+            comment_id: 1000,
+            author: userId,
+            created_at: now,
+            votes: 0,
+            body: newComment,
+          },
+          ...currentComments,
+        ];
+      });
+      const addComment = async () => {
+        const response = await postComment(article_id, {
+          username: userId,
+          body: newComment,
+        });
+        if (response.request.status !== 201) {
+          setComments((currentComments) => {
+            const commentsCopy = [...currentComments];
+            commentsCopy.shift();
+            return commentsCopy;
+          });
+          setSubmitStatus(
+            "Issue with posting comments, please try again later"
+          );
+        } else setSubmitStatus("Comment successfully posted");
+      };
+      addComment();
     }
-    else return (
+  };
+
+  if (isLoading === true) {
+    return <p>Loading...</p>;
+  } else if (isLoading === true && comments.length === 0) {
+    return (
       <section className="comments-section">
+        <p>No comments yet...</p>
+      </section>
+    );
+  } else
+    return (
+      <section className="comments-section">
+        <article className="comment-card">
+          <h3>Post a comment:</h3>
+          <form onSubmit={handleSubmit}>
+            <input onChange={handleChange} placeholder="New comment"></input>
```

Comment:
Input is a single line text field - maybe have a google whether there is something a bit more fitting that allows you to use enter for linebreaks, for example :)